src/Picasso_FieldManager.hpp

```diff
@@ -104,6 +105,34 @@ class FieldManager
         _fields.emplace( key, handle );
     }
 
+    // Uniform mesh constructor.
+    template <class M = Mesh>
+    FieldManager(
+        const std::shared_ptr<M>& mesh,
+        typename std::enable_if<is_uniform_mesh<M>::value>::type* = 0 )
```

Comment:
Shouldn't the uniform and adaptive mesh constructors now be the same? They should have the same interfaces so we should only need a single constructor.

Reply:
Then, can I delete one of the constructor?

Reply:
So, AdaptiveMesh automatically compute buildNodes inside constructor and call `nodes()`, and UniformMesh compute nodes() when it is necessary. So, there are different implement inside the constructor of FieldManager. For AdaptiveMesh it is,

`handle->array = _mesh->nodes();`

For UniformMesh it is 
```
auto cell_size = _mesh->cellSize();
using execution_space =
    typename mesh_type::memory_space::execution_space;
handle->array = _mesh->nodes( cell_size, execution_space() );
```

Wouldn't this matter?

Reply:
Or, should we make UniformMesh same as AdpativeMesh, where buildNodes are computed inside the constructor? Sam and I thought `UniformMesh` need to compute nodes when it is only necessary.

Reply:
I meant we could consider having two constructors, one which matches `AdaptiveMesh` and one that does not build nodes for cases that they are not needed